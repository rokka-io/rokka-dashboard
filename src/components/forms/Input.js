import PropTypes from 'prop-types'
import React from 'react'

const Input = ({ value, defaultValue, placeholder, disabled, ...props }) => {
  let val = disabled || !props.onChange ? value || defaultValue : value
  if (val === null || val === undefined) {
    val = ''
  }

  const ph = defaultValue
    ? `${placeholder || props.name || props.type || ''} (default: ${defaultValue})`
    : placeholder

  return <input {...props} value={val} placeholder={ph} disabled={disabled || !props.onChange} />
}

Input.propTypes = {
  value: PropTypes.any,
  defaultValue: PropTypes.any,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  type: PropTypes.string,
  name: PropTypes.string
}

export default Input
