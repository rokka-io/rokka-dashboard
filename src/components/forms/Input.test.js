import React from 'react'
import Input from '../../../src/components/forms/Input'
import renderer from 'react-test-renderer'

test('Input does render', () => {
  const component = renderer.create(<Input onChange={() => {}} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input does render with a value', () => {
  const component = renderer.create(<Input value="test" onChange={() => {}} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input does forward props', () => {
  const component = renderer.create(<Input foo="bar" onChange={() => {}} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input takes only value if not disabled', () => {
  const component = renderer.create(<Input defaultValue="DefaultValue" />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input takes defaultValue if disabled', () => {
  const component = renderer.create(<Input defaultValue="DefaultValue" disabled />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input shows placeholder', () => {
  const component = renderer.create(<Input placeholder="Placeholder" onChange={() => {}} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input shows placeholder with default value', () => {
  const component = renderer.create(
    <Input defaultValue="DefaultValue" placeholder="Placeholder" onChange={() => {}} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input takes name as placeholder if no placeholder given', () => {
  const component = renderer.create(
    <Input defaultValue="DefaultValue" name="Name" onChange={() => {}} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input takes type as placeholder if no placeholder & name given', () => {
  const component = renderer.create(
    <Input defaultValue="DefaultValue" type="Type" onChange={() => {}} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input is disabled if there is no onChange func', () => {
  const component = renderer.create(<Input />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
