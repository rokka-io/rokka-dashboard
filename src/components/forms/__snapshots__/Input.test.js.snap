// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input does forward props 1`] = `
<input
  disabled={false}
  foo="bar"
  onChange={[Function]}
/>
`;

exports[`Input does render 1`] = `
<input
  disabled={false}
  onChange={[Function]}
/>
`;

exports[`Input does render with a value 1`] = `
<input
  disabled={false}
  onChange={[Function]}
  value="test"
/>
`;

exports[`Input is disabled if there is no onChange func 1`] = `
<input
  disabled={true}
/>
`;

exports[`Input shows placeholder 1`] = `
<input
  disabled={false}
  onChange={[Function]}
  placeholder="Placeholder"
/>
`;

exports[`Input shows placeholder with default value 1`] = `
<input
  disabled={false}
  onChange={[Function]}
  placeholder="Placeholder (default: DefaultValue)"
/>
`;

exports[`Input takes defaultValue if disabled 1`] = `
<input
  disabled={true}
  placeholder=" (default: DefaultValue)"
  value="DefaultValue"
/>
`;

exports[`Input takes name as placeholder if no placeholder given 1`] = `
<input
  disabled={false}
  name="Name"
  onChange={[Function]}
  placeholder="Name (default: DefaultValue)"
/>
`;

exports[`Input takes only value if not disabled 1`] = `
<input
  disabled={true}
  placeholder=" (default: DefaultValue)"
  value="DefaultValue"
/>
`;

exports[`Input takes type as placeholder if no placeholder & name given 1`] = `
<input
  disabled={false}
  onChange={[Function]}
  placeholder="Type (default: DefaultValue)"
  type="Type"
/>
`;
